### PR TITLE
Create a makeshift pillow recipe

### LINKED ISSUE #49031: Feature request: Straw (and similar) pillows
There are currently two type of pillows: The ones filled with downs and the ones filled with cotton.

It would be great if there are pillows which can be filled with straw and other similar fillings. As stated in the [wikipedia article about pillows](https://en.wikipedia.org/wiki/Pillow#Construction_and_parts), one of the possible fillings is straw. [Apparently you can also stuff buckwheat in pillows](https://en.wikipedia.org/wiki/Buckwheat#Upholstery_filling).

I am doing a wilderness run (rural mapgen and city size set to 0) and downs/cotton are very hard to find but straw and withered plants are not. A possible recipe would be a few rags, straw/withered plants/pine boughs and a bit of thread/plant fiber/sinews/... to sew it shut.

It is of course not as comfortable as a "real" pillow but better than nothing. A simple makeshift pillow would also be way easier to make in bulk for e.g your followers.

### PR DESCRIPTION
**Pull request by ActualDio**
#### Summary
Content "Adds a recipe for a makeshift pillow with straw inside it"

#### Purpose of change
Fixes  #49031

#### Describe the solution
Added a recipe to make a makeshift pillow with 4 rags and either 3 straw piles, 15 withered plants or 10 pine bough. I also added a deconstruction recipe that returns 4 rags and 12 withered plants. 
Balance wise I have no idea if this is good or not, will need help on that

#### Testing
Have not tested since at the moment I cant run the game.

#### Additional context

Not needed

### CODE REVIEW

File: data/json/items/generic.json [hunk 1]
```diff
@@ -2388,6 +2388,22 @@
     "flags": [ "SLEEP_AID" ],
     "category": "other"
   },
+  {
+    "type": "GENERIC",
+    "id": "makeshift_pillow",
+    "symbol": ",",
+    "color": "blue",
+    "name": { "str": "makeshift pillow" },
+    "description": "A pillowcase with straw and other dead plants as filling. Its better than nothing...",
```
Comment (pehamm) [inline data/json/items/generic.json:2397]:
```suggestion
    "description": "A pillowcase with straw and other dead plants as filling.  Its better than nothing…",
```
Formatting.

**Comment by pehamm** [comment]
I believe that if set have reversible to true in the recipe, it automatically creates a disassembly recipe that returns the components used to create it in the first place.

Also, in the summary  you need to move the quotation marks to the second part like this:
Content "Adds a recipe for a makeshift pillow with straw inside it"

File: data/json/items/generic.json (hunk 1 above)
Comment (LaVeyanFiend) [inline data/json/items/generic.json:2397]:
```suggestion
    "description": "A pillowcase with straw and other dead plants as filling.  It's better than nothing…",
```

**Comment by faefux** [comment]
Could rags also not be used also as a filler in place of straw, etc. Themselves?

**Pull request merged**